### PR TITLE
Create a jar with all dependencies

### PR DESCRIPTION
**Pull request by mohlendo**
Adds the maven assembly plugin in order to create a jar that contains all the dependencies, but excluding the android ones.

### CODE REVIEW

**Comment by moagrius** [comment]
thank you!

**Comment by Shusshu** [comment]
why do you need an assembly plugin?
Why using old libs - just use https://github.com/mosabua/maven-android-sdk-deployer

File: pom.xml [hunk 1]
```diff
@@ -9,20 +9,20 @@
 	<packaging>jar</packaging>
 	<dependencies>
 		<dependency>
-			<groupId>android</groupId>
+			<groupId>com.google.android</groupId>
 			<artifactId>android</artifactId>
-			<version>4.2_r1</version>
+			<version>4.1.1.4</version>
```
Comment (Shusshu) [inline pom.xml:14]: Old

File: pom.xml [hunk 2]
```diff
@@ -9,20 +9,20 @@
...
+			<artifactId>disklrucache</artifactId>
+			<version>1.3.1</version>
 		</dependency>
 		<dependency>
-			<groupId>android.support</groupId>
-			<artifactId>compatibility-v4</artifactId>
-			<version>12</version>
+			<groupId>com.google.android</groupId>
+			<artifactId>support-v4</artifactId>
+			<version>r7</version>
```
Comment (Shusshu) [inline pom.xml:25]: very bad idea, the latest release (12) should be used
use : maven android sdk deployer https://github.com/mosabua/maven-android-sdk-deployer

File: pom.xml [hunk 3]
```diff
@@ -43,7 +43,22 @@
 				</configuration>
 				<extensions>true</extensions>
 			</plugin>
-
+			<plugin>
+				<artifactId>maven-assembly-plugin</artifactId>
```
Comment (Shusshu) [inline pom.xml:47]: why do you need this plugin ?

When you include the mapview in your project it will automatically include the others

**Comment by moagrius** [comment]
as a non-maven user i have no idea what's going on here.  when this is decided please LMK what the resolution is (roll back, new merge, nothing...?)

**Comment by Shusshu** [comment]
I don't think a rollback is necessary,
if the guy wants a uber jar then he will need the assembly plugin but I certainly won't be using it but it can stay.

My problem is more with the dependencies as it is now using old ones and that shouldn't be the case.

Let me explain.
Maven has a central repository (search.maven.org) and normally google is supposed to upload it's product into that repository but they are not doing it currently.
The android4maven group of volunteers did it for google in the past but they didn't add the latest sdk's and extra's.

Luckily for us Android Maven developers there is an alternative by using Android Maven SDK Deployer which can also be used in combination with a "company" repository.

**Comment by moagrius** [comment]
ok, thanks - just post back if anything's required of me.

i appreciate your help.

**Comment by Shusshu** [comment]
Yes, first I want a reply from the committer and then we'll do the changes

Benoit Billington

**Comment by mohlendo** [comment]
Since the downloadable jar also includes the disklrucache stuff, I thought it would be a good idea to create that with maven. I don't use maven in my project but wanted to use the latest code and so I thought it would be nice to create the uber-jar with maven.

I thought that the dependencies I used, are the newest ones - well, they are the newest on maven central - but actually they are quite old.

**Comment by Shusshu** [comment]
The downloadable jar is for non maven users.
For maven users this is not necessary.

I strongly suggest you to take a look at Maven Android SDK Deployer.

I will submit a new pull request to put other dependencies using the latest
version of google extra & sdk

Benoit Billington

**Pull request merged**